File: sdk/src/main/scala/com/horizen/SidechainNodeViewBase.scala
package com.horizen

import com.horizen.block.{SidechainBlockBase, SidechainBlockHeaderBase}
import com.horizen.chain.AbstractFeePaymentsInfo
import com.horizen.node.{NodeHistoryBase, NodeMemoryPoolBase, NodeStateBase, NodeWalletBase}
import com.horizen.transaction.Transaction

trait SidechainNodeViewBase[
  TX <: Transaction,
  H <: SidechainBlockHeaderBase,
  PM <: SidechainBlockBase[TX, H],
  FPI <: AbstractFeePaymentsInfo,
  NH <: NodeHistoryBase[TX, H, PM, FPI],
  NS <: NodeStateBase,
  NW <: NodeWalletBase,
  NP <: NodeMemoryPoolBase[TX]] {
  def getNodeHistory: NH

  def getNodeState: NS

  def getNodeMemoryPool: NP

  def getNodeWallet: NW

}

File: sdk/src/main/java/io/horizen/SidechainNodeViewBase.java
package io.horizen;

import io.horizen.block.SidechainBlockBase;
import io.horizen.block.SidechainBlockHeaderBase;
import io.horizen.chain.AbstractFeePaymentsInfo;
import io.horizen.node.NodeHistoryBase;
import io.horizen.node.NodeMemoryPoolBase;
import io.horizen.node.NodeStateBase;
import io.horizen.node.NodeWalletBase;
import io.horizen.transaction.Transaction;

public interface SidechainNodeViewBase<
        TX extends Transaction,
        H extends SidechainBlockHeaderBase,
        PM extends SidechainBlockBase<TX, H>,
        FPI extends AbstractFeePaymentsInfo,
        NH extends NodeHistoryBase<TX, H, PM, FPI>,
        NS extends NodeStateBase,
        NW extends NodeWalletBase,
        NP extends NodeMemoryPoolBase<TX>> {
    NH getNodeHistory();

    NS getNodeState();

    NP getNodeMemoryPool();

    NW getNodeWallet();

}